adafruit_ssd1305.py

```diff
@@ -41,6 +41,13 @@
 except ImportError:
     import adafruit_framebuf as framebuf
 
+try:
+    from digitalio import DigitalInOut
```

Comment:
We need to put the typing import as the first one in this try block:
```python
try:
    # Used only for typing
    from typing import Optional
    from digitalio import DigitalInOut
    from busio import I2C, SPI
except ImportError:
    pass
```
The typing import coming first make this raise the exception before any of the other imports occur, which will end up saving some RAM when it runs on microcontrollers.

Reply:
Done! Good call, that makes sense!

Reply:
I liked the comment about the typing imports too, so I added that! Thanks!